tests/0105-transactions_mock.c

```diff
@@ -2866,6 +2866,67 @@ static void do_test_disconnected_group_coord(rd_bool_t switch_coord) {
 }
 
 
+/**
+ * @brief Test that transaction state different from the initial one,
+ * because of a local timed out, doesn't cause an illegal state transition.
+ */
+static void do_test_txn_state_different_from_the_initial_one() {
```

Comment:
```suggestion
static void do_test_txn_state_different_from_the_initial_one(void) {
```

Empty arg-list is an undefined arg-list in C, as opposed to C++ where it is indeed empty.